src/video/cocoa/SDL_cocoamodes.m

```diff
@@ -122,19 +122,24 @@
 static Uint32
 GetDisplayModePixelFormat(CGDisplayModeRef vidmode)
 {
-    /* This API is deprecated in 10.11 with no good replacement (as of 10.15). */
-    CFStringRef fmt = CGDisplayModeCopyPixelEncoding(vidmode);
+    CFStringRef fmt;
     Uint32 pixelformat = SDL_PIXELFORMAT_UNKNOWN;
-
+    CFDictionaryRef dict = (CFDictionaryRef)*((int64_t *)vidmode + 2);
```

Comment:
Relying on implementation-dependent internals like this seems like it'll cause trouble in the future.